engine/src/main/java/org/terasology/engine/GameEngine.java

```diff
@@ -28,6 +28,23 @@
 public interface GameEngine extends AutoCloseable {
 
     /**
+     * @return The current, fine-grained status of the engine.
+     */
+    EngineStatus getStatus();
```

Comment:
Hmmmm... I re-realized getState() is about GameStates and getStatus() is about engine status. And on top of that there are the isRunning/isShuttingDown/isDisposed methods which inform about an engine state that is neither a GameState and an EngineStatus.

How about:

* getState()
* getSubState()
* getGameState()

and do away with the is*() methods?